lib/bolt/logger.rb

```diff
@@ -5,8 +5,11 @@
 module Bolt
   module Logger
     LEVELS = %w[trace debug info notice warn error fatal].freeze
-    @mutex = Mutex.new
-    @warnings = Set.new
+
+    @mutex            = Mutex.new
+    @warnings         = Set.new
+    @disable_warnings = Set.new
```

Comment:
Nick made a good point that we're basically treating this module as a global singleton. I think that's probably OK here - it needs to be global since multiple classes need to access these variables, and all need to access the same instance of those variables. We need to be very careful that accessing the variables is thread-safe though, so any reference to any of these variables needs to be wrapped in a `@mutex.synchronize`.

I'd leave a comment above these variables explaining that we're treating the module like a global singleton so that multiple classes can add warnings.

We also might eventually want to make these their own class that can be instantiated if this gets too hard to test or track, but I think for now this is simple enough we don't need that.

Reply:
Honestly since all of the messages we queue follow a single path to get here, I think it's simpler to just keep passing them through Config to the CLI to here, and avoid having global vars that have to be synched and introduce risk.

Reply:
I disagree. `Bolt::Logger` is already treated as a global singleton due to the `warn_once`/`deprecation_warning` methods and the `@warnings`/`@disable_warnings` variables. In my opinion, passing around log messages between multiple methods in multiple classes adds more complexity by making the code more difficult to understand and maintain in the long-term. Adding an extra `@mutex.synchronize` block around the new methods seems really low risk considering we already do that in this module. 🤷

Reply:
:mask: Ok, can you add a comment though?